src/coprocessor/statistics/cmsketch.rs

```diff
@@ -44,13 +46,42 @@ impl CmSketch {
         }
     }
 
+    pub fn sub(&mut self, bytes: &[u8], cnt: u32) {
+        self.count -= cnt;
+        let (h1, h2) = CmSketch::hash(bytes);
+        for (i, row) in self.table.iter_mut().enumerate() {
+            let j = (h1.wrapping_add(h2.wrapping_mul(i as u64)) % self.width as u64) as usize;
+            row[j] = row[j].saturating_sub(cnt);
+        }
+    }
+
+    pub fn push_to_top_n(&mut self, b: Vec<u8>, cnt: u64) {
+        self.top_n.push((b, cnt))
+    }
+
     pub fn into_proto(self) -> tipb::CmSketch {
         let mut proto = tipb::CmSketch::default();
-        let mut rows = vec![tipb::CmSketchRow::default(); self.depth];
-        for (i, row) in self.table.iter().enumerate() {
-            rows[i].set_counters(row.to_vec());
-        }
+        let rows: Vec<tipb::CmSketchRow> = self
+            .table
+            .into_iter()
+            .map(|row| {
+                let mut pb_row = tipb::CmSketchRow::default();
+                pb_row.set_counters(row);
+                pb_row
+            })
+            .collect();
         proto.set_rows(rows.into());
```

Comment:
```suggestion
        proto.set_rows(rows);
```

Reply:
Since `RepeatedField` implemnts `FromIteratior`, I think this can be used to reduce allocation.